testsuite/features/step_definitions/navigation_steps.rb

```diff
@@ -17,10 +17,8 @@
 end
 
 When(/^I click on "([^"]+)" for "([^"]+)"$/) do |link, item|
-  within(:xpath, '//section') do
-    within(:xpath, "//table/tbody/tr[.//a[contains(.,'#{item}')]]") do
-      find_link(link).click
-    end
+  within(:xpath, "//tr[td[contains(.,'#{item}')]]") do
+    click_link_or_button(link)
```

Comment:
The name of this step is a bit too vague, but I have no better idea...

Reply:
What about:

    When(/^I click on "([^"]+)" for row "([^"]+)"$/)

Reply:
Changed